inc/class_parser.php

```diff
@@ -1709,18 +1709,27 @@ function text_parse_message($message, $options=array())
 
 		$find = array(
 			"#\[(b|u|i|s|url|email|color|img)\](.*?)\[/\\1\]#is",
+			"#\[(email|color|size|font|align|video|)=[^]]*\](.*?)\[/\\1\]#is",
 			"#\[img=([1-9][0-9]*)x([1-9][0-9]*)\](\r\n?|\n?)(https?://([^<>\"']+?))\[/img\]#is",
 			"#\[url=((?!javascript)[a-z]+?://)([^\r\n\"<]+?)\](.+?)\[/url\]#si",
 			"#\[url=((?!javascript:)[^\r\n\"<&\(\)]+?)\](.+?)\[/url\]#si",
 		);
 
 		$replace = array(
+			"$2",
 			"$2",
 			"$4",
 			"$3 ($1$2)",
 			"$2 ($1)",
 		);
-		$message = preg_replace($find, $replace, $message);
+		
+		$messageBefore = "";
+		// The counter limit for this "for" loop is for defensive programming purpose only. It protects against infinite repetition. 
+		for($cnt = 1; $cnt < 5 && $message != $messageBefore; $cnt++)
```

Comment:
If I understand this correctly, the loop has been introduced to remove nested MyCodes. However by limiting it to 5 repetitions, if you have 6+ nested MyCodes there will still be some remaining. Therefore, I do not feel this is the correct way to do it.

Reply:
Yes, you are right, so what do you suggest? The easiest way is to increase the number, but I am not sure if you will like it.

Reply:
I'm also not sure what the best approach would be, personally. This will only do 4 repetitions though (the loop starts at 1, and goes to less than 5 - `1, 2, 3, 4`. 4 repititions seems like an odd limitation.